pyccel/parser/semantic.py

```diff
@@ -925,14 +927,27 @@ def _handle_kernel(self, expr, func, args, **settings):
                         severity='fatal')
             # TODO : type check the NUMBER OF BLOCKS 'numBlocks' and threads per block 'tpblock'
             if not isinstance(expr.numBlocks, LiteralInteger):
-                errors.report("Invalid Block number parameter for Kernel call",
+                try:
+                    numBlocks = self.scope.variables[expr.numBlocks]
```

Comment:
This is not a good way to search for variables as it will fail if the variable is renamed due to collisions. You should use `self.scope.find(expr.numBlocks, category= 'variables')`.

Reply:
Actually there is a function for this which also takes care of raising the error. See `get_variable`

Reply:
Thank you for pointing that out. Fixed e5d7ba0ed13698404e6914ea4ea1b945f8d06c15